youtube_dl/extractor/infomaniak.py

```diff
@@ -0,0 +1,50 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+import json
+
+
+class InfomaniakVOD2IE(InfoExtractor):
+    _VALID_URL = r'https?://player\.vod2\.infomaniak\.com/embed/(?P<id>[^/?#&]+)'
+    _TEST = {
+        'url': 'https://player.vod2.infomaniak.com/embed/1jhvl2uq7kr4y',
+        'md5': 'd06fb3fc5a8d7cb4d6e4a0f4e7c5a76a',
+        'info_dict': {
+            'id': '1jhvl2uq7kr4y',
+            'ext': 'mp4',
+            'title': 'RolandCarey2016-05-04.mp4',
+            'thumbnail': 'https://res.vod2.infomaniak.com/1/vod/thumbnail/1jhvl2uq8yqqv.jpg',
+            'duration': 221,
+        }
+    }
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        # no useful data in given url, formatted url below reveals json data
+        data_url = "https://res.vod2.infomaniak.com/{}/vod/share/{}".format(video_id[0], video_id)
+        webpage = self._download_webpage(data_url, video_id)
+
+        data = json.loads(webpage)['data']['media'][0]
+
+        url = data['source']['url']
+        title = data['title']
+        thumbnail = data.get('image').get('url')
+        duration = data.get('duration')
+
+        video_mimetype = data.get('source').get('mimetype')
+
+        info_dict = {
+            'id': video_id,
+            'url': url,
+            'title': title,
+            'thumbnail': thumbnail,
+            'duration': duration,
+        }
+
+        # if file is m3u8
+        if video_mimetype == 'application/x-mpegurl':
+            info_dict['protocol'] = 'm3u8_native'
+            info_dict['manifest_url'] = url
```

Comment:
Although this will work to some extent, you can find all the formats offered by the manifest. `manifest_url` will be set automatically, I hope.
```suggestion
            info_dict['formats'] = self._extract_m3u8_formats(
                info_dict.pop('url'), video_id, ext='mp4', entry_protocol='m3u8_native', m3u8_id='hls')
            self._sort_formats(info_dict['formats'])
```